.github/pull_request_template.md

```diff
@@ -1,27 +1,29 @@
 # Objective
 
-- Describe the objective or issue this PR addresses.
-- If you're fixing a specific issue, say "Fixes #X".
+Describe the objective or issue this PR addresses.
 
-## Solution
+If you're fixing a specific issue, say "Fixes #X" and the linked issue will automatically be closed when this PR is merged.
+Under each bullet point, describe how this change addressed those objectives if it is not obvious.
 
-- Describe the solution used to achieve the objective above.
+**Changes that will affect external library users must update RELEASES.md before they will be merged.**
 
----
+## Context
 
-## Changelog
+Discuss any context that may be needed for a user with only passing acquaintance with this library to understand the changes you've made.
+This may include related issues, previous discussion, or relevant bits of how the library works).
 
-> This section is optional. If this was a trivial fix, or has no externally-visible impact, you can delete this section.
+## Feedback wanted
 
-- What changed as a result of this PR?
-- If applicable, organize changes under "Added", "Changed", or "Fixed" sub-headings
-- Stick to one or two sentences. If more detail is needed for a particular change, consider adding it to the "Solution" section
-  - If you can't summarize the work, your change may be unreasonably large / unrelated. Consider splitting your PR to make it easier to review and merge!
+> This section is optional. If there are no particularly tricky or controversial changes, you can delete this section.
+
+Which parts of this PR were you unsure about? Which parts were particularly tricky?
+
+If you're stuck on part of the changes or want feedback early, open a draft PR and list the items that need to be completed here using a checklist.
```

Comment:
I think the checklist are not a widely known feature, so linking to documentation might be good here.
```suggestion
If you're stuck on part of the changes or want feedback early, open a draft PR and list the items that need to be completed here using a [checklist](https://github.blog/2014-04-28-task-lists-in-all-markdown-documents/).
```